packages/fxa-auth-server/lib/payments/stripe.ts

```diff
@@ -2288,9 +2291,34 @@ export class StripeHelper {
     // subscriptions for the customer.  The Stripe API stopped including
     // subscriptions for a customer in version 2020-08-27; this ensures
     // backwards compatibility for our code that's relying on that behavior.
+    if (this.stripeFirestore) {
+      if (resourceType === CUSTOMER_RESOURCE) {
+        // @ts-ignore
+        return this.stripeFirestore.retrieveAndFetchCustomer(resource);
+      } else if (resourceType === SUBSCRIPTIONS_RESOURCE) {
+        // @ts-ignore
+        return this.stripeFirestore.retrieveAndFetchSubscription(resource);
+      } else if (resourceType === INVOICES_RESOURCE) {
+        try {
+          const invoice = await this.stripeFirestore.retrieveInvoice(resource);
+          // @ts-ignore
+          return invoice;
+        } catch (err) {
+          if (err.name === FirestoreStripeError.FIRESTORE_INVOICE_NOT_FOUND) {
+            await this.stripeFirestore.retrieveAndFetchCustomer(resource);
+            const invoice = await this.stripe.invoices.retrieve(resource);
+            await this.stripeFirestore.insertInvoiceRecord(invoice);
+            // @ts-ignore
+            return invoice;
+          }
+          throw err;
+        }
+      }
+    }
+
     if (resourceType === CUSTOMER_RESOURCE) {
```

Comment:
I guess once we enable StripeFirestore permanently, we can get rid of the rest of the code from here down...

Reply:
Depends on if the hassle of getting a local firestore working with this correctly outweighs the code penalty.

Reply:
A note, it looks like the local emulator does support the features, I'll add an issue to setup authFirestore to use the local emulator when in local dev mode.